src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -33,11 +34,19 @@ public class HtmlHelper<TModel> : HtmlHelper, IHtmlHelper<TModel>
                 htmlEncoder,
                 urlEncoder)
         {
+            if (expressionTextCache == null)
+            {
+                throw new ArgumentNullException(nameof(expressionTextCache));
+            }
+            ExpressionTextCache = expressionTextCache;
         }
 
         /// <inheritdoc />
         public new ViewDataDictionary<TModel> ViewData { get; private set; }
 
+        /// <inheritdoc />
+        public ExpressionTextCache ExpressionTextCache { get; }
```

Comment:
Just use a `private readonly` field. Can't use `/// <inheritdoc />` because it's not part of the interface but that's minor. No reason (so far) to expose this.